x-pack/plugins/aiops/public/hooks/use_cancellable_request.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useCallback, useRef, useState } from 'react';
+import {
+  type IKibanaSearchResponse,
+  isCompleteResponse,
+  isErrorResponse,
+} from '@kbn/data-plugin/common';
+import { tap } from 'rxjs/operators';
+import { useAiopsAppContext } from './use_aiops_app_context';
+
+export function useCancellableRequest() {
```

Comment:
this function name isn't obvious that it's running a search. 
`useCancellableSearch` might be clearer?

Reply:
renamed in https://github.com/elastic/kibana/pull/144093/commits/b068cebdbf7b21b36bb907bfe7f3d2df47b0cba0